tests/models/test_taskinstance.py

```diff
@@ -537,10 +525,10 @@ def reschedule():
         session.commit()
 
         ti.task = task
-        if state in [State.FAILED, State.UP_FOR_RETRY]:
+        if func.__name__ == "_failure":
```

Comment:
```suggestion
        if func is type(self)._failure:
```

(I think this should work. If not, we can put the function outside of the class instead.)

Reply:
Another possibility would be to add yet another parameter:

```python
def run_expect_catch(ti):
    with pytest.raises(AirflowException):
        ti.run()

def run(ti):
    ti.run()

@pytest.mark.parametrize(
    "..., run_func",
    [
        (..., failure, run_expect_catch),
        (..., success, run),
        ...,
    ],
)
```

Reply:
Thanks @uranusjr, `if func is type(self)._failure:` worked. Adding it now.